pkgs/development/libraries/science/math/ecos/default.nix

```diff
@@ -0,0 +1,37 @@
+{ stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  name = "ecos-${version}";
+  version = "2.0.6";
+
+  src = fetchFromGitHub {
+    owner = "embotech";
+    repo = "ecos";
+    rev = "v${version}";
+    sha256 = "11v958j66wq30gxpjpkgl7n3rvla845lygz8fl39pgf1vk9sdyc7";
+  };
+
+  doCheck = true;
+
+  buildPhase = ''
+    make all shared
+  '';
+
+  checkPhase = ''
+    make test
+    ./runecos
+  '';
+
+  installPhase = ''
+    mkdir -p $out/lib
+    cp lib*.a lib*.so $out/lib
+    cp -r include $out/
+  '';
+
+  meta = with stdenv.lib; {
+    description = "A lightweight conic solver for second-order cone programming";
+    homepage = https://www.embotech.com/ECOS;
+    license = licenses.gpl3;
+    maintainers = [ maintainers.bhipple ];
+  };
```

Comment:
Please specify `platforms` here as well

Reply:
Added `platforms.all`; this is a simple C library with no dependencies, so I'd expect it to "just work" everywhere.  That said, I've only tested it on `x64_64-linux` (and OfBorg seems to have tested it successfully on `aarch64-linux`).